src/Compilers/CSharp/Portable/CommandLine/CSharpCompiler.cs

```diff
@@ -43,30 +43,38 @@ public override Compilation CreateCompilation(TextWriter consoleOutput, TouchedF
 
             var sourceFiles = Arguments.SourceFiles;
             var trees = new SyntaxTree[sourceFiles.Length];
-            var normalizedFilePaths = new String[sourceFiles.Length];
+            var normalizedFilePaths = new string[sourceFiles.Length];
+            var diagnosticBag = DiagnosticBag.GetInstance();
 
             if (Arguments.CompilationOptions.ConcurrentBuild)
             {
                 Parallel.For(0, sourceFiles.Length, UICultureUtilities.WithCurrentUICulture<int>(i =>
                 {
                     //NOTE: order of trees is important!!
-                    trees[i] = ParseFile(consoleOutput, parseOptions, scriptParseOptions, ref hadErrors, sourceFiles[i], errorLogger, out normalizedFilePaths[i]);
+                    trees[i] = ParseFile(parseOptions, scriptParseOptions, ref hadErrors, sourceFiles[i], diagnosticBag, out normalizedFilePaths[i]);
                 }));
             }
             else
             {
                 for (int i = 0; i < sourceFiles.Length; i++)
                 {
                     //NOTE: order of trees is important!!
-                    trees[i] = ParseFile(consoleOutput, parseOptions, scriptParseOptions, ref hadErrors, sourceFiles[i], errorLogger, out normalizedFilePaths[i]);
+                    trees[i] = ParseFile(parseOptions, scriptParseOptions, ref hadErrors, sourceFiles[i], diagnosticBag, out normalizedFilePaths[i]);
                 }
             }
 
             // If errors had been reported in ParseFile, while trying to read files, then we should simply exit.
             if (hadErrors)
             {
+                Debug.Assert(diagnosticBag.HasAnyErrors());
+                ReportErrors(diagnosticBag.ToReadOnlyAndFree(), consoleOutput, errorLogger);
                 return null;
             }
+            else
+            {
+                Debug.Assert(!diagnosticBag.HasAnyErrors());
```

Comment:
@AlekseyTs Never mind, I just found the IsEmptyWithoutResolution property. I'm blind, apparently.